JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -220,8 +220,14 @@ module Microsoft.ApplicationInsights {
         /**
          * helper method to set userId and sessionId cookie
          */
-        public static setCookie(name, value) {
-            Util.document.cookie = name + "=" + value + ";path=/";
+        public static setCookie(name, value, domain?) {
+            var domainAttrib = "";
+
+            if (domain) {
+                domainAttrib = ";domain=" + domain;
```

Comment:
In case domain is not specified should we default to top level domain of the place where the code is executing. So, do something like this: 

`new RegExp(/[-\w]+\.(?:[-\w]+\.xn--[-\w]+|[-\w]{3,}|[-\w]+\.[-\w]{2})$/i).exec(window.location.host)`
I got the regex from here (http://stackoverflow.com/a/6449501). 
Thinking more about it, this would be a bit too risky - we might break a lot of user counting if we get this wrong.